client/common/file.c

```diff
@@ -893,6 +896,12 @@ BOOL freerdp_client_populate_settings_from_rdp_file(rdpFile* file, rdpSettings*
 		free(host);
 	}
 
+	if (~((size_t) file->GatewayAccessToken))
```

Comment:
better use the `!= NULL` syntax (don't know for which reason this strange one has been used there.

Reply:
don't know either, but using != NULL creates a segmentation fault (on x86_64 linux)

Reply:
@jaccobraat Ok, think I found the reason, it is some value was actually set detection coded into the variable